app/controllers/user_controller.rb

```diff
@@ -20,18 +20,16 @@ def signup
   def update_user
     current_user = User.find(params[:id])
     payload = JSON.parse(request.body.read)
-
-    current_user.first_name = payload["first_name"]
-    current_user.email = payload["email"]
-    current_user.user_type = payload["last_name"]
-    current_user.user_type = payload["password"]
-    current_user.user_type = payload["user_type"]
+    current_user.first_name = payload["first_name"] unless payload['first_name']&.empty?
+    current_user.email = payload["email"] unless payload['email']&.empty?
+    current_user.last_name = payload["last_name"] unless payload['last_name']&.empty?
+    current_user.user_type = payload["user_type"] unless payload['user_type']&.empty?
```

Comment:
The [compact method](https://apidock.com/rails/Hash/compact) might be helpful here.

Reply:
Thanks @Li-Hsaun for the feedback. We had to introduce our own customer compact because the inbuilt compact works with nil whereas we are filtering empty string.